euphonic/cli/dispersion.py

```diff
@@ -30,83 +22,42 @@ def main(params: List[str] = None):
         modes = data
         split_args = {'btol': args.btol}
         x_tick_labels = None
-    else:
-        raise TypeError("Input data must be phonon modes or force constants.")
+    modes.frequencies_unit = args.energy_unit
 
     print("Mapping modes to 1D band-structure")
     if args.reorder:
         modes.reorder_frequencies()
 
     spectrum = modes.get_dispersion()
-    if x_tick_labels is not None:
+
+    if args.y_label is None:
+        y_label = f"Energy / {spectrum.y_data.units:~P}"
+    else:
+        y_label = args.y_label
+
+    if x_tick_labels:
```

Comment:
I went with `is not None` so that `[]` would be considered "an empty list of labels" and still replace the default values (which I think are derived using symmetry?).

But this is academic really: the only place x_tick_labels can come from is Seekpath and it will always find at least Gamma and X points. Even if we allowed user-defined points, the range edges should be labelled _somehow_.

As we do not need to deal with `''`, `[]` etc. it doesn't matter whether we check truth or `not None` so this is fine (and easier to read).